tools/autograd/load_derivatives.py

```diff
@@ -85,11 +87,147 @@ def create_derivative(f: NativeFunction, formula: str, var_names: Tuple[str, ...
 
     return Derivative(
         formula=formula,
+        original_formula=original_formula,
         var_names=var_names,
         saved_inputs=saved_inputs,
         saved_outputs=saved_outputs,
     )
 
+def create_forward_derivative(f: NativeFunction, formula: str, names: Tuple[str, ...]) -> ForwardDerivative:
+    assert len(names) == 1, "Forward derivatives can define gradients for only one output at a time"
```

Comment:
Is this a case of multi-output forward codegen not being implemented? That's not the impression I get from the arguments

Reply:
No, it works, but you need to add one line per output.
For example, in backward, you can have a line that does `  input, weight, bias: backward_formula_for_all_inputs()`. This is not allowed for forward formulas.